src/Data/Text.hs

```diff
@@ -361,6 +361,8 @@ instance Read Text where
 -- | @since 1.2.2.0
 instance Semigroup Text where
     (<>) = append
+    stimes = replicate . P.fromIntegral
```

Comment:
I'm somewhat cautious about potential overflow in `fromIntegral`. Let's throw an error if the `Text` is non-empty and `n` does not fit into `Int`, same as `base` do for `ByteArray`: https://hackage.haskell.org/package/base-4.19.1.0/docs/src/Data.Array.Byte.html#stimesPolymorphic

(We do not need to check that `len * n` fits into `Int`, because this is validated by `Data.Text.replicate` itself)

Reply:
I do not follow why we should evaluate to the empty text when given a negative number but evaluate to an error when given a number that is bigger than `maxBound ∷ Int`, but your wish is my command.

Reply:
However, we are not going to check the input in the same way as `base` because we should stay abstract. The way integers are represented is an implementation detail that had changed in the past and may change in the future. Supporting all versions of GHC will need a lot of CPP without significant improvement in performance. So, I shall do an equality check instead of matching on the constructors of `Integer`.

Reply:
> I do not follow why we should evaluate to the empty text when given a negative number but evaluate to an error when given a number that is bigger than maxBound ∷ Int, but your wish is my command.

Ah, I did not remember this peculiarity of `replicate`. 

I'd be in favor of `stimes` throwing an error on negative arguments, even if `replicate` does not mind to swallow it silently. @Lysxia how do you feel about it?

Reply:
Sounds good to me.

Reply:
This is the behaviour on lists:

```
ghci> replicate (-1) [1..10]
[]
```

```
ghci> import Data.Semigroup
ghci> stimes (-1) [1.. 10]
*** Exception: stimes: [], negative multiplier
```

So, `stimes` is not defined even while `replicate` is defined on lists and negative numbers. We can do the same. Let me patch it up.

Reply:
@Bodigrim   So why would you be in favour of `stimes` throwing an error on negative arguments?

@Lysxia   So why does this sound good to you?

I should like to have the underlying reasoning recorded for posterity.

Reply:
Actually, this is a shocking revelation — the property checks of `text` overall compare the behaviour of functions on `Text` to their correspondents on `String`, but **the check for `stimes` is missing** both for strict and lazy `Text`. It would be consistent to add such a check for `stimes` for lazy `Text` and adjust the definitions as needed for it to pass. @Bodigrim Should I add a check for lazy `Text` and make it pass, or should I only add a check for strict `Text`?

Reply:
1. Asking to "replicate a string n times" with negative n is nonsense. There must be an error in the definition of n, so throwing an exception lets users be aware of that error and fix it.
2. The default definition of `stimes` already throws an exception for `n <= 0`. People haven't complained about it. Extending the definition for `n = 0` is reasonable for a monoid.
3. There could still be a case made in favor of making `stimes` less partial and more similar to `replicate` (I think "replicate a string n times" is nonsense as a sentence in natural language, but I don't have a strong argument that code must follow natural language). Until someone makes a good case for extending `stimes`, throwing an exception for negative arguments is forward-compatible: we can extend the function later (it would only break code that catches the exception, a fishy thing to do). If we made it total now and changed our minds later, that would be a more breaking change.

You can add the `stimes` test for strict `Text` now and add lazy `stimes` in another PR (small PR = good!)